### ISSUE TITLE
Document Tanzu cli install without the shell script

### ISSUE THREAD

**Issue by thesteve0** [vmware-tanzu/community-edition#3475]
## Description
Can we please document the manual steps to install the Tanzu CLI. While the shell scripts are nice sometimes they do things a user might not want (for example putting the tanzu binary in /usr/local/bin). It would be good if we could document 
1) Download the binary
2) Run the initialization scripts
3) Any other tasks that should follow

**Comment by kcoriordan** [comment]
@thesteve0 
are you referring to the method shown here in the contributing guide?
https://tanzucommunityedition.io/docs/latest/contribute/contributing/#fetch-the-source

**Comment by stmcginnis** [comment]
FWIW, I don't think it's a good idea to give users instructions on how to do their own custom install. We have a recommended location for the binary, and need other files in the correct place, so adding instructions and giving the impression that it's OK to move things wherever you like isn't a safe thing to do.

For those technically savvy enough, they can just take a look at the `install.sh` file to see what we are doing and try making changes. But then it's explicitly on them that they are deviating from the norm.

**Comment by thesteve0** [comment]
Hey Folks:
I see the Sean has some pretty strong complaints and I am sympathetic to them. Before Kay goes and tries to doc this perhaps we should discuss it a bit before she moves forward. 

I think my main objection now is that 
1) Without prompting the user, tt automatically removes a tanzu binary if it finds one in place
2) If there is a home bin directory it installs there by default but if not it tries to elevate privs and then put in /usr/local/bin

I see all the work the script is going but the TKG install is just
https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.5/vmware-tanzu-kubernetes-grid-15/GUID-install-cli.html#install-the-tanzu-cli-2 
And would actually give the user more control over what happens. 

Basically it comes down to, because of all the automagic, there is a potential to really mess things up for a user when running install.sh if they already have a tanzu cli version installed.

**Comment by dvonthenen** [comment]
> FWIW, I don't think it's a good idea to give users instructions on how to do their own custom install. We have a recommended location for the binary, and need other files in the correct place, so adding instructions and giving the impression that it's OK to move things wherever you like isn't a safe thing to do.

This right here. The other HUGE problem is that how you install everything changes between each and every release (if you look at the install script there is already commented out forward-looking code to handle Tanzu Framework 0.17.0). Most of the time it takes me like 2-4 days just to understand where the heck stuff needs to go in order for everything to install correctly. I would have a strong position that if the install isn't done by how it's prescribed by the install script, the configuration isn't supported.

@thesteve0 for your points:

> Without prompting the user, tt automatically removes a tanzu binary if it finds one in place

The reason why it's like this is for the upgrade scenario. Since the binary is in the path, it's pretty much assumed that's the tanzu cli that you would use to runs commands with. Running the install scripts where a previous installation exists will upgrade (or in the very least "reset") the installation to that prescribed configuration. Please see https://github.com/vmware-tanzu/community-edition/issues/3597

To your point, we could warn the user saying "hey, we detected a previous install, do you want to install this version?" and also have a silent install for automation, scripting, etc. I plan on doing that in a PR to help address https://github.com/vmware-tanzu/community-edition/issues/3597

> If there is a home bin directory it installs there by default but if not it tries to elevate privs and then put in /usr/local/bin

The `$home/bin` is a safe not global space to install the CLI. Otherwise, the prescribed location according to TKG is `/usr/local/bin`. We are just following TKG's prescribed way of install.


> Basically it comes down to, because of all the automagic, there is a potential to really mess things up for a user when running install.sh if they already have a tanzu cli version installed.

The automatic is a good thing when has it not been? Most people don't care how it gets installed, they just want it installed correctly. I feel the install script does *exactly* what it needs to do (with some needed warnings perhaps - again will have a PR for that shortly). If the install script doesn't do exactly what it's supposed to do, that's a bug. I would be infinitely more concerned with users attempting to roll their own "new" installation method each and every release.

**Comment by dvonthenen** [comment]
PTAL https://github.com/vmware-tanzu/community-edition/pull/4211
